src/Planner/PlannerContext.h

```diff
@@ -56,18 +57,18 @@ class PlannerSet
 {
 public:
     /// Construct planner set that is ready for execution
-    explicit PlannerSet(SetPtr set_)
+    explicit PlannerSet(FutureSet set_)
         : set(std::move(set_))
     {}
 
     /// Construct planner set with set and subquery node
-    explicit PlannerSet(SetPtr set_, QueryTreeNodePtr subquery_node_)
-        : set(std::move(set_))
+    explicit PlannerSet(QueryTreeNodePtr subquery_node_)
+        : set(promise_to_build_set.get_future())
```

Comment:
Where `promise_to_build_set` should be assigned?

Reply:
If I understand the question correctly, the answer is: promise_to_build_set is default-constructed and after this it has a valid internal state so we can get a future from it.

Reply:
And it's one set in CreatingSet via `promise_to_fill_set.set_value`, got it